refnx/analysis/curvefitter.py

```diff
@@ -907,6 +907,9 @@ def bounds_list(parameters):
                 isinstance(param.bounds, Interval)):
             bnd = param.bounds
             bounds.append((bnd.lb, bnd.ub))
+        elif (hasattr(param, 'bounds') and isinstance(param.bounds, PDF) and
+              param.bounds.rv.a != -np.inf and param.bounds.rv.b != -np.inf):
```

Comment:
PDF can be created with any class that has `logpdf` and `rvs` methods. So perhaps this code block is better written as:

```
        elif (hasattr(param, 'bounds') and isinstance(param.bounds, PDF) and isinstance(param.bounds.rv, (scipy.stats.rv_continuous, scipy.stats._distn_infrastructure.rv_frozen)) and np.isfinite([param.bounds.rv.a, param.bounds.rv.b]).all()):
```

assuming it works.